### PR TITLE
Add interaction controller for interactive transitions

### PR DESCRIPTION
**Pull request by mrgrauel**
### What does this PR do?

* adds `Interactr` class which can easily be used with gesture recognizers to implement interactive transitions.

### Presentation
```swift
let translation = sender.translation(in: view)
let progress = interactr.calculateProgress(translation, viewBounds: view.bounds, direction: .right)
        
interactr.mapGestureState(sender.state, progress: progress) {
    customPresentViewController(presentr, viewController: toViewController, animated: true, completion: nil)
}
```

### Dismiss
```swift
let translation = sender.translation(in: view)
let progress = interactr.calculateProgress(translation, viewBounds: view.bounds, direction: .left)
        
interactr.mapGestureState(sender.state, progress: progress) { [unowned self] in
    self.dismiss(animated: true, completion: nil)
}
```

### CODE REVIEW

File: Presentr/Interactr.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+//  Presentr
+//
+//  Created by Andreas Grauel on 26.07.17.
+//  Copyright © 2017 danielozano. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public class Interactr: UIPercentDrivenInteractiveTransition {
```
Comment (gabrielPeart) [inline Presentr/Interactr.swift]: @mrgrauel nice catch

File: Presentr/Interactr.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        case left
+        case right
+    }
+    
+    var hasStarted = false
+    var shouldFinish = false
+    
+    public var percentThreshold: CGFloat = 0.3
+    
+    public func calculateProgress(_ translationInView:CGPoint, viewBounds:CGRect, direction:Direction) -> CGFloat {
```
Comment (gabrielPeart) [inline Presentr/Interactr.swift]: @mrgrauel try to be consistent. On the line 23 you added one style for the type (adding a space), now you didn't do it. Keep the same standard all over your code.

Comment (mrgrauel) [inline Presentr/Interactr.swift]: I did it cause it's a public variable, the others are internal. How would @gabrielPeart like to have it?

**Comment by mrgrauel** [comment]
Did you push your changes your talked about? I can't find any new commits.

File: Presentr/Interactr.swift (hunk 2 above)
Comment (gabrielPeart) [inline Presentr/Interactr.swift]: @mrgrauel How come code indentation is correlated to access modifiers? 
I'm talking about adding the fact that you added an extra space after the colon in some cases and not in others

**Comment by gabrielPeart** [comment]
I didn't have the time to work on it. Having a really busy period now.

**Comment by mrgrauel** [comment]
@gabrielPeart it's updated. Let me know if you want any other changes.

**Comment by mrgrauel** [comment]
@gabrielPeart any updates?

**Comment by danlozano** [comment]
I'll take a look at this soon!

Sorry for the wait.

**Comment by mrgrauel** [comment]
@danlozano again a week past. Any news?

**Comment by mrgrauel** [comment]
@danlozano almost three weeks past. Any news?

**Comment by danlozano** [comment]
Please add an example of how this can be used/why it's useful/etc... in the example project so I can take a further look into it.

Thanks

**Comment by mrgrauel** [comment]
Sorry, but the response comes to late we found another solution.

**Comment by danlozano** [comment]
No worries!

**Pull request closed without merging**